### DISCUSSION TITLE
Installation failure with 4.8.0-0.okd-2021-10-24-061736

### DISCUSSION THREAD

**Discussion by dav1dli** [okd-project/okd#956] [Q&A]
**Describe the bug**
UPI on KVM/QEMU VMs fails

**Version**
4.8.0-0.okd-2021-10-24-061736

**How reproducible**
Follow the documented procedure up to the boot of created nodes. Bootstrap boots from ISO, installs the filesystem, reboots and goes into configuration. It is available on the network and can be SSHed using the configtred SSH public key.   Last entries ` journalctl -b -f -u release-image.service -u bootkube.service` show:
```
Oct 27 15:43:44 okd4-bstr bootkube.sh[11642]: Skipped "secret-kube-apiserver-to-kubelet-signer.yaml" secrets.v1./kube-apiserver-to-kubelet-signer -n openshift-kube-apiserver-operator as it already exists
Oct 27 15:43:44 okd4-bstr bootkube.sh[11642]: Skipped "secret-loadbalancer-serving-signer.yaml" secrets.v1./loadbalancer-serving-signer -n openshift-kube-apiserver-operator as it already exists
Oct 27 15:43:44 okd4-bstr bootkube.sh[11642]: Skipped "secret-localhost-serving-signer.yaml" secrets.v1./localhost-serving-signer -n openshift-kube-apiserver-operator as it already exists
Oct 27 15:43:44 okd4-bstr bootkube.sh[11642]: Skipped "secret-service-network-serving-signer.yaml" secrets.v1./service-network-serving-signer -n openshift-kube-apiserver-operator as it already exists
```
When other nodes are booted from the same ISO, `sudo coreos-installer install --ignition-url=http://web-server.example.com/master.ign /dev/sda --insecure-ignition` is executed they install the filesystem, reboot and go in an endless loop with messages 
```
ignition GET error https://api-int.okd4.example.com:22623/config/master: dial tcp: lookup api-int.okd4.example.com on [::1]:53 read udp [::1]:52330->[::1]:53" read: connection refused
```
It looks like if the network is not up. Command `curl -k https:// api-int.okd4.example.com:22623/config/master | jq` from any other host on the network produces an expected JSON.
I tried to configure static IPs and add kernel parameters 
```
coreos.inst=yes
coreos.inst.install_dev=sda
coreos.inst.image_url=http://web-server.example.com/fedora-coreos.raw.xz
coreos.inst.ignition_url=http://web-server.example.com/master.ign
coreos.inst.insecure
rd.neednet=1
ip=10.1.1.10::10.1.1.1:255.255.255.0:okd4-mstr3.okd4.example.com:ens3:none nameserver=10.1.1.24
```
The issue is reproducible 100% of times with all nodes.

**Log bundle**
` ./openshift-install gather bootstrap --dir=./okd4inst` produces
FATAL bootstrap host address and at least one control plane host address must be provided

**Comment by vrutkovs** [comment]
>./openshift-install gather bootstrap --dir=./okd4inst produces
FATAL bootstrap host address and at least one control plane host address must be provided

Can you collect the log bundle when bootstrap IP and all control plane addresses are specified?

**Comment by disposab1e** [comment]
I use for example: (UPI - CentOS 8.4 / KVM/QEMU)
```
kernel http://10.0.0.2/fcos/4-8-0-0-okd-2021-10-24-061736/vmlinuz 
net.ifnames=1 
ip=10.0.0.112::10.0.0.1:255.255.255.0:worker-2.okd.example.com:ens3:none 
nameserver=10.0.0.1 
rd.neednet=1 
coreos.inst=yes 
coreos.inst.install_dev=vda 
console=tty0 
console=ttyS0 
coreos.live.rootfs_url=http://10.0.0.2/fcos/4-8-0-0-okd-2021-10-24-061736/rootfs 
coreos.inst.ignition_url=http://10.0.0.2/ignition/4-8-0-0-okd-2021-10-24-061736/worker.ign 
initrd http://10.0.0.2/fcos/4-8-0-0-okd-2021-10-24-061736/initrd
boot
```
FCOS: 34.20210626.3.1

```
rootfs:
live-rootfs.x86_64.img

vmlinuz:
live-kernel-x86_64

initrd:
live-initramfs.x86_64.img
```
